cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/SubstationIdMapping.java

```diff
@@ -274,27 +355,27 @@ private static Set<String> allConnected(Map<String, Set<String>> adjacency, Set<
         return new HashSet<>(allConnected);
     }
 
-    private static String representativeVoltageLevelId(Collection<String> voltageLevelIds) {
-        return voltageLevelIds.stream()
-                .sorted()
-                .findFirst()
-                .orElseThrow(() -> new IllegalStateException("Unexpected: voltageLevelIds list is empty"));
+    private String representativeVoltageLevelId(Collection<String> voltageLevelIds) {
+        Optional<String> existingVoltageLevelId = voltageLevelIds.stream()
+                .filter(voltageLevelId -> !fictitiousVoltageLevels.containsKey(voltageLevelId))
+                .min(Comparator.naturalOrder());
+        if (existingVoltageLevelId.isPresent()) {
+            return existingVoltageLevelId.get();
+        }
+        Optional<String> fictitiousVoltageLevelIdForLineContainer = voltageLevelIds.stream()
+                .filter(voltageLevelId -> !isSubstationContainer(voltageLevelId)) // Fictitious for LineContainer
+                .min(Comparator.naturalOrder());
+        return fictitiousVoltageLevelIdForLineContainer.orElseGet(() -> voltageLevelIds.stream().min(Comparator.naturalOrder())
+                .orElseThrow(() -> new IllegalStateException("Unexpected: voltageLevelIds list is empty")));
     }
 
     private String representativeSubstationId(Collection<String> substationIds) {
-        return substationIds.stream()
-                .filter(substationId -> context.config().substationIdsExcludedFromMapping()
-                        .stream()
-                        .noneMatch(substationId::matches))
```

Comment:
I know `Conversion.Config.substationIdsExcludedFromMapping()` returns an empty collection, but nor this method or its class are final, so it could be redefined and used:

```java
public Network importData(CgmesModel cgmes) {
    Conversion.Config config = new Conversion.Config() {
        @Override
        public List<String> substationIdsExcludedFromMapping() {
            return List.of("substation_1");
        }
    };
    return new Conversion(cgmes, config).convert();
}
```
Is it something we want to exclude?
- If yes, `Conversion.Config.substationIdsExcludedFromMapping()` should be removed and this should be reported as a breaking change.
- If no, this `filter` clause should be kept.

Reply:
Thanks @marqueslanauja!
I also think it is better to keep it in this PR.